Rac 4090/ha deployment

johren opened the pull request:
Configures the control network, DHCP, /etc/hosts, and hostname on all nodes in the cluster.

Checks the corosync and pacemaker status to verify that we have created a happy cluster.

Had to modify fit_common.remote_shell and fit_common.scp_file_to_ora to pass in a VM number.
Defaults to node 1.

Tested by running the following with 3 and 4 node clusters.
```
python ./run_tests.py -test deploy/os_ova_install.py -stack 4 -template http://mirrors.hwimo.lab.emc.com/mirrors/ova/ha-ubuntu-1604.ova -config dellemc-test/config-mn -numvms 4

python ./run_tests.py -test deploy/rackhd_ha_install.py -config dellemc-test/config-mn -stack 4 -numvms 4
```

hohene commented on test/deploy/rackhd_ha_install.py:
```diff
@@ -0,0 +1,109 @@
+            hosts_conf.write(line)
+        hosts_conf.close()
+
+    def test04_install_hosts_config(self):
```
test04 - here and below, looks like copy/paste issue

johren replied:
Will fix

hohene commented on test/deploy/rackhd_ha_install.py:
```diff
@@ -0,0 +1,109 @@
+
+
+if __name__ == '__main__':
+    fit_common.unittest.main()
```
can get rid of fit_common here  - ie. import unittest above and change to unittest.main

johren replied:
Cut-n-paste.  Will fix.

hohene commented on test/deploy/rackhd_ha_install.py:
```diff
@@ -0,0 +1,109 @@
+            self.assertEqual(fit_common.remote_shell('ping -c 1 -w 5 ' + control_ip, vmnum=vmnum)['exitcode'],
+                             0, 'Control NIC failure.')
+
+    def test02_install_dhcp_config(self):
```
Suggestions: For a deploy script, you could use nosedep and @depends if you want some of the tests to be skipped if the previous one failed.

johren replied:
Great suggestion!  Thanks.

johren commented:
test this please

yyscamper commented:
@johren : please squash your commit history.

hohene commented on test/deploy/rackhd_ha_install.py:
```diff
@@ -0,0 +1,217 @@
+from nosedep import depends
+
+# set path to common libraries
+sys.path.append(subprocess.check_output("git rev-parse --show-toplevel", shell=True).rstrip("\n") + "/test/common")
```
Update this script to use fit_path and gets rid of this sys.path.append line.  (see the test/template file.)

johren replied:
Okay if I fix that in a quick follow-on PR?  I currently don't need to rebase and it would be easier to do a separate PR.

hohene replied:
yep, it should still work as is.

hohene commented on test/deploy/rackhd_ha_install.py:
```diff
@@ -0,0 +1,217 @@
+        rendered = template.render(node_list=self.create_node_list())
+        return rendered
+
+    def test03_install_hosts_config(self):
```
do you want this to depend on prior test results?   if so, add @depends(after=xxxx)

hohene commented on test/deploy/rackhd_ha_install.py:
```diff
@@ -0,0 +1,217 @@
+                             0, "Hosts Config failure.")
+        os.remove('hosts-conf')
+
+    def test04_install_hostname_config(self):
```
same comment - add the @depend(after=xxx) if you want the test to be skipped.  Same with subsequent test05, 06, 07.

johren commented:
Squashed and rebased and then realized that part of the rebase was changing scp_file_to_ora to scp_file_to_host.  I have incorporated that change into rackhd_ha_install.py and I am currently testing that change.

johren commented:
@hohene @gpaulos @larry-dean @patelb10

johren commented:
test this please
